examples/server/server.cpp

```diff
@@ -3783,6 +3795,48 @@ int main(int argc, char ** argv) {
         return 0;
     });
 
+    if (state.load() == SERVER_STATE_ERROR) {
```

Comment:
This check is not needed, because `listen_after_bind` is already in another thread. By the time main thread reaches this code, `listen_after_bind` in the other thread may not yet finish executing.

Reply:
Your are right and the check is also just right before now.